modules/fedimint-ln-client/src/pay.rs

```diff
@@ -258,6 +283,28 @@ impl LightningPayFunded {
     async fn gateway_pay_invoice(
         gateway: LightningGateway,
         payload: PayInvoicePayload,
+    ) -> Result<String, GatewayPayError> {
+        let mut last_error = None;
+
+        // If we don't succeed to reach the gateway in 30 seconds we'd rather abort the
+        // payment so the user isn't surprised by a heavily delayed payment
+        for _ in 0..30 {
```

Comment:
One a slow network this could take a lot longer then 30 seconds. Would it be better to just have ` loop` wraped in a `timeout()`?

Reply:
Good idea!